### PR TITLE
API: StepScan interface, a DataBroker than returns DataFrames

### PR DESCRIPTION
**Pull request by danielballan**
Like DataBroker, you can do `StepScan[...]` and `StepScan.find_headers(...)`. Instead of returning a Header, it gets the header(s), fetches all Events, builds one Muxer, and returns `DataMuxer.to_sparse_dataframe()`.

If the Headers have asynchronous Events or unlike EventDescriptors, `StepScan` raises.

This is an experimental feature, seeking user feedback.

### CODE REVIEW

File: dataportal/scans/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from stepscan import StepScan
```
Comment (tacaswell) [inline dataportal/scans/__init__.py]: I thought we were just going to go with `scans.py` for simplicity at this point, but meh

**Comment by ericdill** [comment]
Is this is failing because it depends on #134?

**Comment by ericdill** [comment]
Generally :+1: to this PR. Some minor quibbles about error messages, but nothing worth holding this PR up.

**Comment by danielballan** [comment]
@ericdill Yes.

File: dataportal/scans/__init__.py (hunk 1 above)
Comment (danielballan) [inline dataportal/scans/__init__.py]: If you want access to StepScan the class and StepScan the singleton, this is the cleanest way to do that.

Comment (tacaswell) [inline dataportal/scans/__init__.py]: as discussed in person, the class and the singleton should have different names.

**Pull request merged**